.github/workflows/release.yml

```diff
@@ -0,0 +1,315 @@
+          DEBIAN_FRONTEND=noninteractive apt-get -y --no-install-recommends install \
+              build-essential \
+              ca-certificates \
+              curl \
+              doxygen \
+              g++ \
+              gdb \
+              git \
+              gnupg \
+              gpg \
+              less \
+              libboost-all-dev \
+              libc-ares-dev \
+              libre2-dev \
+              libssl-dev \
+              ninja-build \
+              pkg-config \
+              software-properties-common \
+              sudo \
+              wget \
+              zlib1g-dev
+
+          # Add the public key for the llvm repository to get the correct clang version
+          bash -c 'wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key|apt-key add -'
+          apt-add-repository -y 'deb http://apt.llvm.org/focal/ llvm-toolchain-focal-15 main'
+
+          # Add public key and repository to get cmake 3.25+
+          wget -O - https://apt.kitware.com/keys/kitware-archive-latest.asc 2>/dev/null | gpg --dearmor - > /usr/share/keyrings/kitware-archive-keyring.gpg
+          echo 'deb [signed-by=/usr/share/keyrings/kitware-archive-keyring.gpg] https://apt.kitware.com/ubuntu/ focal main' > /etc/apt/sources.list.d/kitware.list
+
+          apt-get update
+          apt-get -y --no-install-recommends install -t llvm-toolchain-focal-15 clang-15 clang-tidy-15
+          apt-get -y install cmake
+
+          mkdir -p ${HOME}/opt/src
+          cd ${HOME}/opt/src && \
+            git clone --recurse-submodules -b v1.52.0 --depth 1 --shallow-submodules https://github.com/grpc/grpc && \
+            cd grpc && \
+            mkdir -p build && \
+            cd build && \
+            cmake .. -G Ninja \
+                -DgRPC_ZLIB_PROVIDER=package \
+                -DgRPC_CARES_PROVIDER=package \
+                -DgRPC_RE2_PROVIDER=package \
+                -DgRPC_SSL_PROVIDER=package \
+                -DgRPC_INSTALL=ON \
+                -DgRPC_BUILD_TESTS=OFF \
+                -DBUILD_SHARED_LIBS=ON \
+                -DCMAKE_BUILD_TYPE=RelWithDebInfo \
+                -DCMAKE_INSTALL_PREFIX=/usr/local \
+                -DCMAKE_INSTALL_RPATH=/usr/local/lib && \
+            ninja install && \
+            cd .. && \
+            rm -rf build
+
+      - name: Setup build directory
+        run: mkdir builds
+
+      - name: Build
+        shell: bash
+        run: |
+          mkdir build
+          pushd build/
+          /usr/bin/cmake .. -G Ninja
+          ninja all -j 3 && ninja install
+          popd
+
+      - name: Copy
+        run: |
+          cp build/install/lib/libviam_rust_utils.a builds/libviam_rust_utils-${{ matrix.platform }}.a
+          cp build/install/lib/libviamapi.so.${{ needs.prepare.outputs.version }} builds/libviamapi-${{ matrix.platform }}.so
+          cp build/install/lib/libviamsdk.so.${{ needs.prepare.outputs.version }} builds/libviamsdk-${{ matrix.platform }}.so
```

Comment:
Are these binaries functional without the gRPC binaries from `/usr/local`?

Reply:
I see, good to know! In that case perhaps we should scrap the whole "release binary" aspect of this PR entirely and just have the "release" job include source code. It still lets us have specific versioned releases to point folks to but will then require people to build from source.

Reply:
It has been a while, but I believe that if you create a tag and then make a github "release" of that tag, it will automatically make a source tarball of it.

Reply:
Also, to clarify: you still can make binary releases if you think it is useful, but you can't do it if you are building any of the dependencies (like grpc) yourself. So you probably could make releases for MacOS, Bookworm, Bullseye, and Jammy across various target CPUs. I just don't think you can do Focal.

Reply:
Good to know, thanks!